[Material Dialogs] Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention.

galiak11 opened the pull request:
[Material Dialogs] Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention.

Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention.

Issue: b/148428512.

material-automation commented:
bazel detected changes to the following targets:

```
//catalog:MDCCatalog
//catalog:MDCCatalog.apple_binary
//catalog:MDCCatalog.swift_runtime_linkopts
//catalog:MDCCatalogLib
//components/Dialogs:ColorThemer
//components/Dialogs:DialogThemer
//components/Dialogs:Dialogs
//components/Dialogs:ObjcExamples
//components/Dialogs:SwiftExamples
//components/Dialogs:Theming
//components/Dialogs:TypographyThemer
//components/Dialogs:private
//components/Dialogs:snapshot_test_lib
//components/Dialogs:snapshot_tests
//components/Dialogs:snapshot_tests.swift_runtime_linkopts
//components/Dialogs:snapshot_tests_test_binary
//components/Dialogs:snapshot_tests_test_bundle
//components/Dialogs:unit_test_sources
//components/Dialogs:unit_test_swift_sources
//components/Dialogs:unit_tests
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3.swift_runtime_linkopts
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_binary
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3.swift_runtime_linkopts
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_binary
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Dialogs:unit_tests_environment
//components/Dialogs:unit_tests_environment.swift_runtime_linkopts
//components/Dialogs:unit_tests_environment_test_binary
//components/Dialogs:unit_tests_environment_test_bundle
```

<!-- identifier: affected-targets -->

material-automation commented:
## API diff detected the following changes

### Dialogs

#### MDCAlertControllerView

*modified* class: `MDCAlertControllerView`

| Type of change: | Declaration |
|---|---|
| From: | `@interface MDCAlertControllerView : UIView  @property(nonatomic, strong, nullable) UIFont *titleFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *titleColor UI_APPEARANCE_SELECTOR;  @property(nonatomic, strong, nullable) UIImage *titleIcon; @property(nonatomic, strong, nullable) UIColor *titleIconTintColor; @property(nonatomic, assign) NSTextAlignment titleAlignment; @property(nonatomic, assign) NSTextAlignment messageAlignment;  /**  An optional custom icon view above the title of the alert.   @note This property is intended to be used to provide a custom implementation of the title icon  view. If the intention is to just display a `UIImage`, use `setTitleIcon:` API instead. If  'titleIcon' is set, 'titleIconView' is ignored.  */ @property(nonatomic, strong, nullable) UIView *titleIconView;  @property(nonatomic, strong, nullable) UIFont *messageFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *messageColor UI_APPEARANCE_SELECTOR;  // b/117717380: Will be deprecated (x3) @property(nonatomic, strong, nullable) UIFont *buttonFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *buttonColor UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *buttonInkColor UI_APPEARANCE_SELECTOR;  @property(nonatomic, assign) CGFloat cornerRadius;  /*  Indicates whether the view's contents should automatically update their font when the device’s  UIContentSizeCategory changes.   This property is modeled after the adjustsFontForContentSizeCategory property in the  UIContentSizeCategoryAdjusting protocol added by Apple in iOS 10.   Default value is NO.  */ @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)     BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;  /**  By setting this property to @c YES, the Ripple component will be used instead of Ink  to display visual feedback to the user.   @note This property will eventually be enabled by default, deprecated, and then deleted as part  of our migration to Ripple. Learn more at  https://github.com/material-components/material-components-ios/tree/develop/components/Ink#migration-guide-ink-to-ripple   Defaults to NO.  */ @property(nonatomic, assign) BOOL enableRippleBehavior;  @end` |
| To: | `@interface MDCAlertControllerView : UIView  @property(nonatomic, strong, nullable) UIFont *titleFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *titleColor UI_APPEARANCE_SELECTOR;  @property(nonatomic, strong, nullable) UIImage *titleIcon; @property(nonatomic, strong, nullable) UIColor *titleIconTintColor; @property(nonatomic, assign) NSTextAlignment titleAlignment; @property(nonatomic, assign) NSTextAlignment messageAlignment;  /**  An optional custom icon view above the title of the alert.   @discussion This property is intended to be used to provide a custom implementation of the title  icon view. If the intention is to just display a `UIImage`, use `setTitleIcon:` API instead. If  this property value is nil, the title icon is displayed above the title view. If you set this  property to a custom view, it is displayed instead of the title icon. Custom title views are  aligned with the title and may be resized to fit.  */ @property(nonatomic, strong, nullable) UIView *titleIconView;  @property(nonatomic, strong, nullable) UIFont *messageFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *messageColor UI_APPEARANCE_SELECTOR;  // b/117717380: Will be deprecated (x3) @property(nonatomic, strong, nullable) UIFont *buttonFont UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *buttonColor UI_APPEARANCE_SELECTOR; @property(nonatomic, strong, nullable) UIColor *buttonInkColor UI_APPEARANCE_SELECTOR;  @property(nonatomic, assign) CGFloat cornerRadius;  /*  Indicates whether the view's contents should automatically update their font when the device’s  UIContentSizeCategory changes.   This property is modeled after the adjustsFontForContentSizeCategory property in the  UIContentSizeCategoryAdjusting protocol added by Apple in iOS 10.   Default value is NO.  */ @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)     BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;  /**  By setting this property to @c YES, the Ripple component will be used instead of Ink  to display visual feedback to the user.   @note This property will eventually be enabled by default, deprecated, and then deleted as part  of our migration to Ripple. Learn more at  https://github.com/material-components/material-components-ios/tree/develop/components/Ink#migration-guide-ink-to-ripple   Defaults to NO.  */ @property(nonatomic, assign) BOOL enableRippleBehavior;  @end` |

<!-- identifier: apidiff -->

iberezan commented:
Please pick up the latest changes from cl/300797913. It was updated

iberezan reviewed (commented):
Please pick up the latest changes from cl/300797913

iberezan committed:
[Material Dialogs] Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention.

Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention.

Issue: b/148428512.
PiperOrigin-RevId: 300797913

galiak11 commented:
Closing because this was implemented in: "PR #9915 - [Dialogs] Custom title icon public API: Flip the priority between title icon and title custom view to mirror UIKit convention."
